File: Frontend/IMMRequest/src/app/models/request.ts
import { AdditionalFieldValue } from './additionalFieldValue';
import { Type } from './type'
import { Guid } from "guid-typescript";

export class Request {
  id: string;
  state: string;
  type: Type;
  date: Date;
  typeId: Guid;
  description: string;
  requestorsName: string;
  requestorsEmail: string;
  requestorsPhone: string;
  additionalFieldValues: AdditionalFieldValue[];

  constructor(id: string, typeId: Guid, description: string, requestorsName: string, type: Type, date: Date,
    requestorsEmail: string, requestorsPhone: string, additionalFieldValues: AdditionalFieldValue[]) {
    this.type = type;
    this.typeId = typeId;
    this.description = description;
    this.requestorsName = requestorsName;
    this.requestorsEmail = requestorsEmail;
    this.requestorsPhone = requestorsPhone;
    this.additionalFieldValues = additionalFieldValues;
  }
}

File: Backend/IMMRequest/IMMRequest.Domain/Entities/Request.cs
using System.Collections.Generic;
using System;

namespace IMMRequest.Domain
{
    public class Request
    {
        public Guid Id { get; set; }
        public DateTime Date { get; set; }
        public string State { get; set; }
        public string RequestorsName { get; set; }
        public string RequestorsEmail { get; set; }
        public string RequestorsPhone { get; set; }
        public string Description { get; set; }
        public TypeEntity Type { get; set; }
        public Guid TypeId { get; set; }
        public ICollection<AdditionalFieldValue> AdditionalFieldValues { get; set; }

        public Request()
        {
            this.Id = Guid.NewGuid();
            this.AdditionalFieldValues = new List<AdditionalFieldValue>();
        }

        public Request(string RequestorsName, string RequestorsEmail, string RequestorsPhone, 
        TypeEntity Type, string Description, List<AdditionalFieldValue> AdditionalFieldValues) 
        {
            this.Id = Guid.NewGuid();
            this.RequestorsName = RequestorsName;
            this.RequestorsEmail = RequestorsEmail;
            this.RequestorsPhone = RequestorsPhone;
            this.Type = Type;
            this.Description = Description;
            this.AdditionalFieldValues = AdditionalFieldValues;
        }
    }
}
